### DISCUSSION TITLE
Running a Workload Simulation for the Firefly App

### DISCUSSION THREAD

**Discussion by ymahajan** [firefly-iii/firefly-iii#9126] [General]
My objective is to simulate real-world usage of the Firefly app to assess its performance, stability, and scalability under load. The workload should represent typical user interactions, including browsing, data entry, querying, and other specific operations.

I would like to check if the Firefly app provides built-in benchmarking tools or scripts designed for performance testing. These tools often offer pre-configured scenarios to test various aspects of the application. Also would like to ensure the dataset used in the simulation is representative of production data to provide meaningful insights.

**Comment by JC5** [comment]
It doesn't have these. There is a [data generator](https://codeberg.org/firefly-iii/data-generator) on Codeberg, which contains the demo site data suite, you could use that.

**Comment by ymahajan** [comment]
Thanks @JC5. We have tested the Firefly III app on [YugabyteDB](https://docs.yugabyte.com/) and it works fine. We can provide the instruction set to install YugabyteDB, which should be simple. How do you suggest we should put in a request to update relevant documentation on your end?

**Reply by JC5** [reply]
Sorry, but I'm not sure why I should. Your [Laravel demo application](https://docs.yugabyte.com/preview/drivers-orms/php/laravel/) runs as `DB_CONNECTION=pgsql` which is well documented already.
